homeassistant/components/unifiprotect/migrate.py

```diff
@@ -2,153 +2,141 @@
+@callback
+def check_if_used(
+    hass: HomeAssistant, entry: ConfigEntry, entities: dict[str, EntityRef]
+) -> dict[str, EntityUsage]:
+    """Check for usages of entities and return them."""
+
+    entity_registry = er.async_get(hass)
+    refs: dict[str, EntityUsage] = {
+        ref: {"automations": {}, "scripts": {}} for ref in entities
+    }
+
+    for entity in er.async_entries_for_config_entry(entity_registry, entry.entry_id):
+        for ref_id, ref in entities.items():
+            if (
+                entity.domain == ref["platform"]
+                and entity.disabled_by is None
+                and ref["id"] in entity.unique_id
+            ):
+                entity_automations = automations_with_entity(hass, entity.entity_id)
+                entity_scripts = scripts_with_entity(hass, entity.entity_id)
+                if entity_automations:
+                    refs[ref_id]["automations"][entity.entity_id] = entity_automations
+                if entity_scripts:
+                    refs[ref_id]["scripts"][entity.entity_id] = entity_scripts
+
+    return refs
 
-    registry = er.async_get(hass)
-    to_migrate = []
-    for entity in er.async_entries_for_config_entry(registry, entry.entry_id):
-        if entity.domain == Platform.BUTTON and "_" not in entity.unique_id:
-            _LOGGER.debug("Button %s needs migration", entity.entity_id)
-            to_migrate.append(entity)
-
-    if len(to_migrate) == 0:
-        _LOGGER.debug("No button entities need migration")
-        return
-
-    count = 0
-    for button in to_migrate:
-        device = bootstrap.get_device_from_id(button.unique_id)
-        if device is None:
-            continue
-
-        new_unique_id = f"{device.id}_reboot"
-        _LOGGER.debug(
-            "Migrating entity %s (old unique_id: %s, new unique_id: %s)",
-            button.entity_id,
-            button.unique_id,
-            new_unique_id,
-        )
-        try:
-            registry.async_update_entity(button.entity_id, new_unique_id=new_unique_id)
-        except ValueError:
-            _LOGGER.warning(
-                "Could not migrate entity %s (old unique_id: %s, new unique_id: %s)",
-                button.entity_id,
-                button.unique_id,
-                new_unique_id,
+
+@callback
+def create_repair_if_used(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    breaks_in: str,
+    entities: dict[str, EntityRef],
+) -> None:
+    """Create repairs for used entities that are deprecated."""
+
+    usages = check_if_used(hass, entry, entities)
+    for ref_id, refs in usages.items():
+        issue_id = f"deprecate_{ref_id}"
+        automations = refs["automations"]
+        scripts = refs["scripts"]
+        if automations or scripts:
+            items = sorted(
+                set(
+                    chain.from_iterable(
+                        list(automations.values()) + list(scripts.values())
+                    )
+                )
+            )
```

Comment:
```suggestion
            items = sorted(
                set(chain.from_iterable(chain(automations.values(), scripts.values())))
            )
```

I think this should work